src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestPartials.qs

```diff
@@ -1,26 +1,51 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-namespace Microsoft.Quantum.Testing.QIR
-{
+namespace Microsoft.Quantum.Testing.QIR {
     open Microsoft.Quantum.Intrinsic;
 
+    newtype LittleEndian = Qubit[];
+
+    operation Dummy (qs : Qubit[]) : Unit is Adj + Ctl {
+    }
+
+    operation InnerNestedTuple(tuple : (Int, Double), (a1 : String, a2 : Qubit)) : Unit is Adj + Ctl {
+    }
+
+    operation TakesNestedTuple(tuple : (Int, Double), a1 : String, a2 : Qubit) : Unit is Adj + Ctl {
+    }
+
+    operation ApplyToLittleEndian(bareOp : ((Qubit[]) => Unit is Adj + Ctl), register : LittleEndian)
+    : Unit is Adj + Ctl {
+        bareOp(register!);
+    }
+
```

Comment:
Not sure if this is the right place for it but maybe we can also test an operation that takes a partial as an input argument? I believe there are some use cases for it (e.g. [ApplyToEach](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.canon.applytoeach))

Reply:
The tests are not very well structured, sorry. There are tests for that but they are in TestOpArgument.qs.